Feature/2197 replay with input #2197

iamviqar opened the pull request:
<!-- Thanks for submitting a Pull Request to Kestra. To help us review your contribution, please follow the guidelines below:

- Make sure that your commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification e.g. `feat(ui): add a new navigation menu item` or `fix(core): fix a bug in the core model` or `docs: update the README.md`. This will help us automatically generate the changelog.
- The title should briefly summarize the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share a flow example to help the reviewer understand and QA the change.
- Use "closes" to automatically close an issue. For example, `closes #1234` will close issue #1234. -->

### What changes are being made and why?

<!-- Please include a brief summary of the changes included in this PR e.g. closes #1234. -->

---

### How the changes have been QAed?

<!-- Include example code that shows how this PR has been QAed. The code should present a complete yet easily reproducible flow.

```yaml
# Your example flow code here
```

Note that this is not a replacement for unit tests but rather a way to demonstrate how the changes work in a real-life scenario, as the end-user would experience them.

Remove this section if this change applies to all flows or to the documentation only. -->

---

### Setup Instructions

<!--If there are any setup requirements like API keys or trial accounts, kindly include brief bullet-points-description outlining the setup process below.

- [External System Documentation](URL)
- Steps to set up the necessary resources

If there are no setup requirements, you can remove this section.

Thank you for your contribution. ❤️  -->

iamviqar committed:
CONTRIBUTING.md : addind node options variable

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

MilosPaunovic committed:
chore(ui): improve wording of contribution guide

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

iamviqar committed:
Merge branch 'kestra-io:develop' into develop

iamviqar committed:
adding inputs in task replay dialogue

iamviqar committed:
initializing the input form

iamviqar committed:
Merge branch 'kestra-io:develop' into feature/2197-replay-with-input

iamviqar committed:
prefilling the input form

iamviqar committed:
#2197: commiting the progress with pending implemention

Skraye commented on ui/src/components/executions/Overview.vue:
```diff
@@ -6,8 +6,8 @@
                 <set-labels :execution="execution" />
-                <restart is-replay :execution="execution" class="ms-0" @follow="forwardEvent('follow', $event)" />
-                <restart :execution="execution" class="ms-0" @follow="forwardEvent('follow', $event)" />
+                <restart is-replay :initial-inputs="flow" :execution="execution" class="ms-0" @follow="forwardEvent('follow', $event)" />
```
```suggestion
                <restart is-replay :flow="flow" :execution="execution" class="ms-0" @follow="forwardEvent('follow', $event)" />
```

If you're passing a flow, just call it flow, same for replay component

Skraye commented on ui/src/stores/executions.js:
```diff
@@ -24,8 +24,8 @@ export default {
             return this.$http.post(
-                `${apiUrl(this)}/executions/${options.executionId}/restart`,
-                null,
+                `${apiUrl(this)}/executions/${options.executionId}/restart-with-input`,
```
Do not modify the path

Skraye commented on webserver/src/main/java/io/kestra/webserver/controllers/api/ExecutionController.java:
```diff
@@ -794,6 +794,27 @@ public Execution restart(
     }
 
+    @ExecuteOn(TaskExecutors.IO)
+    @Post(uri = "/{executionId}/restart-with-input")
```
Those endpoint (duplicated of the original one for now) are not necessary, just need to enhance the existing one

same apply for the replay endpoint

Skraye reviewed (changes requested):
* Change property name in component
* Do not create new endpoint
* As you're modifying controllers, do not forget to add new tests

More details in the review comments

iamviqar commented:
Thanks, @Skraye , for your review comments! I’ll address them as soon as possible.

iamviqar committed:
Update ui/src/components/executions/Overview.vue

Co-authored-by: YannC <37600690+Skraye@users.noreply.github.com>

iamviqar committed:
#2197 : addressing code review comments with pending backend changes

iamviqar commented:
@Skraye I've addressed most of your code review comments, except for the tests on the backend API. I marked this PR as a draft because there are still pending implementations for the backend API. I'm currently trying to determine how to pass the input values received as parameters in the execution controller to the actual execution. I noticed there aren't any setter methods for inputs in the execution object, and I would appreciate your assistance with this.

Regards
tagging @MilosPaunovic

Skraye commented:
Hello @iamviqar, any update on this?

MilosPaunovic commented:
Hi @iamviqar, are there any updates on this?

anna-geller commented:
@MilosPaunovic @Skraye can one of you take over to finish and merge the PR?

MilosPaunovic commented:
This would have to be @Skraye, as it would involve some Java work also.

Piyush-r-bhaskar commented:
Closing in favour of https://github.com/kestra-io/kestra/pull/11010
